packages/server/src/core/procedure.ts

```diff
@@ -85,7 +86,14 @@ export type ProcedureArgs<TParams extends ProcedureParams> =
  *
  * @internal
  */
-export interface ProcedureBase<TParams extends ProcedureParams> {
+export interface ProcedureBase<
+  TParams extends ProcedureParams,
+  Kind extends string,
```

Comment:
```suggestion
  Kind extends ProcedureType
```

Reply:
Seems like a good idea, but given this is an internal type, I was thinking in my mind that keeping the type constraints looser is actually going to improve perf, since it removes another look-up that actually doesn't affect the static checking on the user's side.

This is very hand-wavvy, though. I would love to find time for benchmarking large complex nested routers in the longer term, because without benchmarks everything I'm saying about performance should be very questionable 😂

Reply:
Oh yea I don't know anything about TS-perf just thought it would perhaps prevent some mistakes if giving it a wrong Kind. I do love my squiggles and autocomplete 😏